### PR TITLE
U: Deleting slack client direct dependencie

### PR DESCRIPTION
**Pull request by kvaillant**
U: Adding tests & dev modules
U: Adding Slack.use function to import a slack client module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,11 +21,12 @@ username: name displayed in the chat channel. default "winston-slack"
 
 
     var winston = require('winston');
+    var slackClient = require('my-favorite-slack-client');
```
Comment (Spidy88) [inline README.md]: What is the point of this?

File: lib/winston-slack.js [hunk 2]
```diff
@@ -1,20 +1,19 @@
 /**
  * Created by andy.batchelor on 2/26/14.
+ * Contributed by karl.vaillant on 28/11/14 -> Deleting node modules direct dependencies which may not be supported yet.
  */
 
-var nodeSlack = require('node-slack'),
-    util = require('util'),
+var util = require('util'),
```
Comment (Spidy88) [inline lib/winston-slack.js]: Not consistent styling. Need a semi-colon on the previous line, or no `var` here. Based on the current style, just remove the `var`

File: README.md (hunk 1 above)
Comment (kvaillant) [inline README.md]: Hi, 

I'm so sorry to answer right now only.

The purpose is to use whatever nodejs module for slack that a developer wants. It's just an example.
One can use https://www.npmjs.com/package/slackhook for exemple

File: lib/winston-slack.js (hunk 2 above)
Comment (kvaillant) [inline lib/winston-slack.js]: I removed line 5, that's why I add `var` here.

File: README.md (hunk 1 above)
Comment (Spidy88) [inline README.md]: That's what forks are for. This library wouldn't be usable right out of the box if the consumer had to select a slack module (and then make that slack module available to this library, i.e. updating the package.json).

File: lib/winston-slack.js (hunk 2 above)
Comment (Spidy88) [inline lib/winston-slack.js]: Totally missed that :+1:

**Pull request closed without merging**